sponge/sponge.ml

```diff
@@ -219,17 +223,25 @@ module Make_hash (P : Intf.Permutation) = struct
     update params ~state:init inputs |> digest
 end
 
+type sponge_state = Absorbed of int | Squeezed of int [@@deriving sexp]
+
+type 'f t =
+  { mutable state: 'f State.t
+  ; params: 'f Params.t
+  ; mutable sponge_state: sponge_state }
+
+let make ~state ~params ~sponge_state = {state; params; sponge_state}
+
 module Make_sponge (P : Intf.Permutation) = struct
   open P
 
+  let make = make
+
   let capacity = 1
 
   type sponge_state = Absorbed of int | Squeezed of int [@@deriving sexp]
```

Comment:
Nit: I think this needs to be aliased to the new `sponge_state` type above, or deleted.